hardware/raspberrypi/spi/README.md

```diff
@@ -3,51 +3,91 @@
 ## Page Contents
 
 - [Overview](#overview)  
-- [Software](#software)
 - [Hardware](#hardware)
+- [Software](#software)
 - [Linux driver](#driver)
 - [Troubleshooting](#troubleshooting)
 
 <a name="overview"></a>
 ## Overview
 
-The Raspberry Pi is equipped with one [SPI](https://en.wikipedia.org/wiki/Serial_Peripheral_Interface_Bus) bus that has 2 chip selects.
-
-The SPI master driver is disabled by default on Raspberry Pi OS. To enable it, use [raspi-config](../../../configuration/raspi-config.md), or ensure the line `dtparam=spi=on` isn't commented out in `/boot/config.txt`, and reboot. If the SPI driver was loaded, you should see the device `/dev/spidev0.0`.
-
-The SPI bus is available on the P1 Header:
-
-```
-MOSI    P1-19
-MISO    P1-21
-SCLK    P1-23   P1-24    CE0
-GND     P1-25   P1-26    CE1
-```
-
-<a name="software"></a>
-## Software
-
-### bcm2835 library
-
-This is a C library for Raspberry Pi (RPi). It provides access to GPIO and other IO functions on the Broadcom BCM 2835 chip. Accesses the hardware registers directly.
-
-http://www.airspayce.com/mikem/bcm2835/
-### Use spidev from C
-
-There's a loopback test program in the Linux documentation that can be used as a starting point. See the [Troubleshooting](#troubleshooting) section. Uses the Linux `spidev` driver to access the bus.
-
-### Shell
-
-```bash
-# Write binary 1, 2 and 3
-echo -ne "\x01\x02\x03" > /dev/spidev0.0
-```
+The Raspberry Pi family of devices is equipped with three [SPI](https://en.wikipedia.org/wiki/Serial_Peripheral_Interface_Bus) buses. The main SPI interface is referred to as SPI0 in the documentation; the second is SPI1. SPI2 is only usable on a Compute Module or Compute Module 3. Devices based on BCM2711, i.e. Pi 4B and the upcoming Compute Module 4, have access to an additional 4 SPI buses (SPI3 to SPI6).
 
 <a name="hardware"></a>
 ## Hardware
 
-The BCM2835 on the Raspberry Pi has 3 SPI Controllers. Only the SPI0 controller is available on the header.
-Chapter 10 in the [BCM2835 ARM Peripherals](../bcm2835/BCM2835-ARM-Peripherals.pdf) datasheet describes this controller.
+The BCM2835 on the Raspberry Pi has 3 [http://en.wikipedia.org/wiki/Serial_Peripheral_Interface_Bus SPI] Controllers. The main SPI bus (with two hardware chip selects) is available on the header of all Pis with Linux kernel support. The second SPI bus (with three hardware chip selects) is available on 40-pin versions of Pis, with kernel support from Raspbian Jessie 2016-05-10 distribution and up. The third SPI bus (also with three hardware chip selects) is only usable on a Compute Module because the pins aren't brought out onto the 40-pin header.
```

Comment:
"... has 3 Controllers" - I'm not 100% sure of the terms, but is it more accurate to say it has 2 SPI controllers and 3 SPI busses?

Reply:
No - there are 3 distinct controllers.

Reply:
Ah, so does that mean that the words "controller" and "bus" are totally interchangeable, or is there still some subtle difference? If they **are** interchangeable, it might be worth using "bus" exclusively and not mention "controller" (if that would help to avoid confusion) ?

Reply:
One could argue that each group of pins where an SPI controller can be mapped constitutes a bus. SPI0 has two _natural_ mappings, 7-11 and 35-39 (although there's nothing preventing a pick-and-mix approach), so one could time share SPI0 between the two buses, albeit to no practical advantage over adding a few more software chip-selects.

Even if there is a one-to-one correspondence, I still think it's useful distinction - the devices are wired to buses, buses are managed by controllers. When I see a bunch of cyclists on the road I don't call them bikes.

Reply:
Sorry for nitpicking, it was just the "...has 3 SPI Controllers. The main SPI bus... The second SPI bus... The third SPI bus..." phrasing that confused me. (And I was assuming that if it confused me, it might confuse other readers too)

Reply:
...but I see that the new wording of this section removes this confusion anyway :+1: